### ISSUE TITLE
[no-unused-vars] More missing detections

### ISSUE THREAD

**Issue by milesj** [typescript-eslint/typescript-eslint#45]
**Repro**

Run the linter on https://github.com/milesj/beemo

Some examples of it not working:

- ~~Type index access:~~ https://github.com/milesj/beemo/blob/master/packages/cli/src/CLI.ts#L58

```
11:32  warning  'DriverContext' is defined but never used    no-unused-vars
```

- ~~When an imported type is re-assigned and exported:~~ https://github.com/milesj/beemo/blob/master/packages/core/src/types.ts#L34 AND https://github.com/milesj/beemo/blob/master/packages/core/src/types.ts#L62

```
6:10  warning  'Tool' is defined but never used          no-unused-vars
7:10  warning  'ExecaReturns' is defined but never used  no-unused-vars
```

- ~~Imported type used in a generic intersection:~~ https://github.com/milesj/beemo/blob/master/packages/driver-babel/src/BabelDriver.ts#L46

```
  8:18  warning  'DriverArgs' is defined but never used  no-unused-vars
  9:10  warning  'BabelArgs' is defined but never used   no-unused-vars
```

- Imported namespace reference: https://github.com/milesj/beemo/blob/master/packages/driver-webpack/src/types.ts#L9

```
  6:8  warning  'webpack' is defined but never used  no-unused-vars
```

**Expected Result**

They don't fail.

**Actual Result**

They do fail.

**Additional Info**

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `0.14.0` |
| `typescript-eslint-parser` | `21.0.2` |
| `typescript`               | `3.2.2` |

**Comment by armano2** [comment]
This is expected, there was a lot of changes in parser, and in not compatible with plugin, you have to wait for release

**Comment by milesj** [comment]
Yup, assumed that was the problem. Will report back in a week.

**Comment by aserra54** [comment]
Not sure if I should piggy-back off of this issue, but another false negative I'm seeing is in interface declarations, e.g.

```
import { Foo, Bar } from 'qux';

export class Baz implements Foo {
  public bar: Bar;
}
```

Both `Foo` and `Bar` get tagged as being unused. Sample here:

https://github.com/aserra54/typescript-eslint-parser-no-unused-vars

**Comment by armano2** [comment]
@aserra54 you are missing plugin https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin

```
"extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
```

**Comment by milesj** [comment]
Tested with the new `@typescript-eslint` packages. All of them are fixed except one, when referencing an imported namespace. For example, `i18n` here.

```ts
import i18next from 'i18next';

export interface Translator extends i18next.i18n {}
```

**Comment by SimenB** [comment]
I'm getting false positives on React components.

```tsx
// react.tsx
import * as React from 'react';
import { Heading2 } from 'some-place';

export const Comp: React.FunctionComponent = () => (
  <Heading2>Some content</Heading2>
);
```

```
/Users/simen/repos/monorepo/react.tsx
  2:10  warning  'Heading2' is defined but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (0 errors, 1 warning)
```

config:

```json
{
  "extends": ["plugin:@typescript-eslint/recommended"],
  "plugins": ["@typescript-eslint"]
}
```

(this creates a couple of more errors for rules I've turned off)

**Comment by armano2** [comment]
You have to install and use eslint plugin React

**Comment by SimenB** [comment]
Ah! Fair enough 🙂 I've forgotten all about that plugin, gonna be awesome to be able to use it again 😅
